operator/config/overlays/openshift/manager_node_selector_patch.yaml

```diff
@@ -0,0 +1,9 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: controller-manager
+spec:
+  template:
+    spec:
+      nodeSelector:
+        kubernetes.io/os: linux
```

Comment:
Does it make sense to add the `nodeSelector` to the base deployment instead of just adding it for OpenShift? This should reduce issues when running on "vanilla Kubernetes" as well when there are non-linux nodes in the cluster.

Reply:
It makes sense but we won't be able to control that. What I mean is if we define `nodeSelector` to the base deployment then we are putting restriction to the users that our loki operator works only with linux which I don't think is the case? Maybe in future we can extend this through a flag or something else? But I guess it is out of scope for this PR right now. WDYT?

Reply:
The way the build is currently set up there are only images for linux and my guess is that nobody has yet tested the operator on any other operating system yet, so I would assume that it only runs on linux.

I have no experience with mixed-OS clusters but I would assume that every deployment needs a nodeSelector on which operating system it is able to run, otherwise the scheduler might place pods on incompatible nodes where the software is then not able to start causing unnecessary errors and, in the worst case, needing manual intervention. So I wouldn't think of this as a restriction but actually a help for normal users.

People that build their own image for running on a different operating system are probably experienced enough to also change the nodeSelector then.

Reply:
So should I update this into base deployment?
cc @periklis

Reply:
I agree here with @xperimental. As long as we officially support linux only for now, we can keep the node selector on the base deployment.